aten/src/ATen/native/special_functions/detail/promote.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+namespace at {
+namespace native {
+namespace special_functions {
+namespace detail {
+template<typename T1, bool = std::is_integral<T1>::value>
+struct promotion {
+  using type = double;
```

Comment:
is it indented to promote integral types to double? In pytorch usually integral types are promoted to default floating type (which, by default, is `float`)

Reply:
But the resulting tensor would still be default floating type, so this would only affect internal computations.